File: ReelEnglish_mobil/fluxlingo_app/lib/models/quiz.dart
class Quiz {
  final String id;
  final String videoId;
  final String question;
  final List<String> options;
  final int correctAnswerIndex;
  final String explanation;

  Quiz({
    required this.id,
    required this.videoId,
    required this.question,
    required this.options,
    required this.correctAnswerIndex,
    required this.explanation,
  });

  factory Quiz.fromJson(Map<String, dynamic> json) {
    return Quiz(
      id: json['id'] ?? '',
      videoId: json['video_id'] ?? '',
      question: json['question'] ?? '',
      options: List<String>.from(json['options'] ?? []),
      correctAnswerIndex: json['correct_answer_index'] ?? 0,
      explanation: json['explanation'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'video_id': videoId,
      'question': question,
      'options': options,
      'correct_answer_index': correctAnswerIndex,
      'explanation': explanation,
    };
  }
}

File: ReelEnglish_backend/internal/models/quiz.go
package models

// Quiz temsil eder
type Quiz struct {
	ID                   string   `json:"id" firestore:"id"`
	VideoID              string   `json:"video_id" firestore:"video_id"`
	Question             string   `json:"question" firestore:"question"`
	Options              []string `json:"options" firestore:"options"`
	CorrectAnswerIndex   int      `json:"correct_answer_index" firestore:"correct_answer_index"`
	Explanation          string   `json:"explanation" firestore:"explanation"`
	DifficultyLevel      string   `json:"difficulty_level" firestore:"difficulty_level"`
	GrammarTopic         string   `json:"grammar_topic" firestore:"grammar_topic"`
}

// QuizSubmission quiz cevabını temsil eder
type QuizSubmission struct {
	QuizID                string `json:"quiz_id"`
	SelectedAnswerIndex   int    `json:"selected_answer_index"`
	UserID                string `json:"user_id"`
	IsCorrect             bool   `json:"is_correct"`
	XPEarned              int    `json:"xp_earned"`
}
